test/base/test_diagitensor.jl

```diff
@@ -508,16 +508,16 @@ using Test
     end
 
     @testset "Contraction (Diag uniform * Diag uniform, all contracted)" begin
-      D1 = δ(l, i, k, j)
-      D2 = δ(j, l, i, k)
+      D1 = δ(l, i, k, j) * 1.0 # hack, convert to Float64
+      D2 = δ(j, l, i, k) * 1.0
```

Comment:
What goes wrong if you don't add these?

Reply:
similar to abiove
```
Test threw exception
  Expression: D2 * D1 ≈ dense(D1) * dense(D2)
  InexactError: Bool(3)
```

Reply:
That is related to an issue in our contract code that it doesn't always handle mixtures of element types gracefully, which is being improved in our `NDTensors` rewrite.